Automattic/wp-calypso
# Site Settings: hook, add and introduce Jetpack Disconnect Site

**This is a reference PR.** Cleaned and moved to #18119

Introduce the Disconnect Site, hook it up and add route definitions. Also, the style is added and included for loading. 

This is a landing PR for the new Jetpack Disconnect Flow. It introduces a basic layout of the first page of the flow and makes it visible from the /settings/disconnect-site page (in dev). 

The data layer for the Disconnect Flow is not yet implemented here, nor is saving the survey data.
Style details (e.g. breakpoints) are not fully refined as not all the details are visible yet, so IMHO it makes sense to finalize them in the follow-up PR.

Dev feature flag included.

**To test**:
- Checkout this branch, or use calypso.live
 https://calypso.live/?branch=add/site-settings-disconnect-site

For Jetpack sites (incl. the dev mode):
 - verify that the start of the flow is visible at ```/settings/disconnect-site/:site```, where ```:site``` is the Jetpack site.

For non-Jetpack and Atomic sites, 
 - verify immediate redirection to the General settings

With empty Redux state we should see a placeholder prior to rendering.

**Depends on**: https://github.com/Automattic/wp-calypso/pull/17555

**Visuals**:
![screen shot 2017-09-07 at 10 44 13](https://user-images.githubusercontent.com/13561163/30154235-8d216862-93b9-11e7-9f2d-7e99096f4261.png)
